encoding/json/decode.go

```diff
@@ -1045,11 +1045,13 @@ func (d *Decoder) decodeType(valueJSON interface{}, results typeDecodingResults)
 		return nil
 	}
 
-	typeID, ok := valueJSON.(string)
-	if ok {
+	if typeID, ok := valueJSON.(string); ok {
 		if result, ok := results[typeID]; ok {
 			return result
 		}
+
+		// Backwards-compatibility for format <0.3.0
+		return cadence.TypeID(typeID)
```

Comment:
Probably need to add this behind a flag, because this is also used for decoding program arguments, which could cause problems

Reply:
Good point, added in [85b3ed3](https://github.com/onflow/cadence/pull/1733/commits/85b3ed329f0fe7769bcb334d50ca64aaed5bd9de)

Reply:
Also added a test case in 942041d98b4055a35a0c0ae7015eeeaf991053e8